platform/FileHandle.h

```diff
@@ -152,8 +162,106 @@ class FileHandle {
      */
     MBED_DEPRECATED_SINCE("mbed-os-5.4", "Replaced by FileHandle::size")
     virtual off_t flen() { return size(); }
+
+    /** Set blocking or non-blocking mode of the file operation like read/write.
+     *  Definition depends upon the subclass implementing FileHandle.
+     *  The default is blocking.
+     *
+     *  @param blocking true for blocking mode, false for non-blocking mode.
+     */
+    virtual int set_blocking(bool blocking) {
+        return -1;
+    }
+
+    /** Check for poll event flags
+     * The input parameter can be used or ignored - the could always return all events,
+     * or could check just the events listed in events.
+     * Call is non-blocking - returns instantaneous state of events.
+     * Whenever an event occurs, the derived class must call _poll_change().
+     * @param events bitmask of poll events we're interested in - POLLIN/POLLOUT etc.
+     * @return bitmask of poll events that have occurred.
+     */
+    virtual short poll(short events) const {
```

Comment:
The `poll` function is growing on me as an extensible approach to handling the state of non-blocking classes, but it is a shift for mbed. I'm thinking this is fine as long as non-blocking behaviour is completely optional. Both implementors and users should be able to ignore these functions.

A few other concerns:
1. We should move the non-class functions and defines out into a seperate file, thoughts on `mbed_poll.h`? alternatively they could live in `mbed_retarget.h`.
1. Can we use `uint16_t` here (and why not `uint32_t`? or maybe even a typedef for `poll_t`? maybe). This is especially important as a bitfield.
1. I would describe the purpose of the `events` parameter a bit better, maybe:
   > The input parameter is a bit mask specifying the events the application is interested in. The input parameter can be ignored, but may be used to avoid the cost of determining event state.

   Also, can you make the comments all share the same style, complete with indention and spacing?
1. There was talk earlier of providing a default implementation of the `sigio` behaviour with a ticker. Is that possible to accomplish here in the attach function as default behaviour?
1. Can we drop the `_poll_change` function? It would seem simpler for implementors to take care of storing the attached callback themselves.